satpy/readers/agri_l1.py

```diff
@@ -154,7 +166,10 @@ def calibrate_to_reflectance(self, data, channel_index, ds_info):
         """Calibrate to reflectance [%]."""
         logger.debug("Calibrating to reflectances")
         # using the corresponding SCALE and OFFSET
-        cal_coef = 'CALIBRATION_COEF(SCALE+OFFSET)'
+        if self.filename_info['platform_id'] == 'FY4B':
+            cal_coef = 'Calibration/CALIBRATION_COEF(SCALE+OFFSET)'
+        else:
+            cal_coef = 'CALIBRATION_COEF(SCALE+OFFSET)'
         num_channel = self.get(cal_coef).shape[0]
```

Comment:
As discussed on slack, there are some optimizations that could be made to this reader. The main one I noticed was this `.get` of the coefficients variable in this method. It does 3 `.get` calls which is creating a DataArray 3 times and on two of them it does `.values` which means they are computed twice. Additionally, the coefficients are used for every channel that is loaded (even if it is just a sub-index). I refactored this into a cached_property locally that I think you could apply here. Here is my diff:

```diff
diff --git a/satpy/readers/agri_l1.py b/satpy/readers/agri_l1.py
--- a/satpy/readers/agri_l1.py	(revision 6a0e6da340023971f4a54c480bfe037b702c9219)
+++ b/satpy/readers/agri_l1.py	(date 1656293104023)
@@ -32,6 +32,7 @@
 
 from satpy.readers._geos_area import get_area_definition, get_area_extent
 from satpy.readers.hdf5_utils import HDF5FileHandler
+from satpy._compat import cached_property
 
 logger = logging.getLogger(__name__)
 
@@ -96,6 +97,12 @@
         """Init filehandler."""
         super(HDF_AGRI_L1, self).__init__(filename, filename_info, filetype_info)
 
+    @cached_property
+    def reflectance_coeffs(self):
+        # using the corresponding SCALE and OFFSET
+        cal_coef = 'CALIBRATION_COEF(SCALE+OFFSET)'
+        return self.get(cal_coef).values
+
     def get_dataset(self, dataset_id, ds_info):
         """Load a dataset."""
         ds_name = dataset_id['name']
@@ -153,14 +160,13 @@
     def calibrate_to_reflectance(self, data, channel_index, ds_info):
         """Calibrate to reflectance [%]."""
         logger.debug("Calibrating to reflectances")
-        # using the corresponding SCALE and OFFSET
-        cal_coef = 'CALIBRATION_COEF(SCALE+OFFSET)'
-        num_channel = self.get(cal_coef).shape[0]
+        coeffs = self.reflectance_coeffs
+        num_channel = coeffs.shape[0]
         if num_channel == 1:
             # only channel_2, resolution = 500 m
             channel_index = 0
-        data.attrs['scale_factor'] = self.get(cal_coef)[channel_index, 0].values.item()
-        data.attrs['add_offset'] = self.get(cal_coef)[channel_index, 1].values.item()
+        data.attrs['scale_factor'] = coeffs[channel_index, 0].item()
+        data.attrs['add_offset'] = coeffs[channel_index, 1].item()
         data = scale(data, data.attrs['scale_factor'], data.attrs['add_offset'])
         data *= 100
         ds_info['valid_range'] = (data.attrs['valid_range'] * data.attrs['scale_factor'] + data.attrs['add_offset'])
```